src/components/map/map-gps-stations-layer.tsx

```diff
@@ -49,15 +53,80 @@ export class MapGPSStationsLayer extends BaseComponent<IProps, IState> {
         />
       );
     });
+    const mapScaleAdjust = 50;
+
+    const velocityArrows = visibleGPSStations.map(stat => {
+      // map scale is in meters per pixel, so use this as the basis for consistent length arrows at different zooms
+      // though the distances are so large we need to reduce the multiplier by a constant to fit the screen.
+      const arrowScale = mapScale / mapScaleAdjust;
+
+      const startLatLng = new LatLng(stat.latitude, stat.longitude);
+      // calculate the magnitude of the two component velocity values
+      const velMag = Math.sqrt((stat.eastVelocity ** 2) + (stat.northVelocity ** 2));
+      // Adjust the scale of the magnitude to improve visibility
+      // Stations move at a rate of meters per year, which is a small number.
+      // so we multiply by the map scale
+      const magnitude = velMag * arrowScale;
+      const dir = Math.atan(stat.northVelocity / stat.eastVelocity);
```

Comment:
This works because the `northVelocity` is almost always positive for these stations, but it would give wrong answers if any happened to head South -- it would rotate it by 180º. I only recently learned about `atan2`, which gives correct directions in all four quadrants of the plain.

```
  x   y         Math.atan(y/x)      Math.atan2(y, x)
  1   1             Pi/4                   Pi/4
  1  -1            -Pi/4                  -Pi/4
 -1   1            -Pi/4                   3/4 Pi
 -1  -1             Pi/4                  -3/4 Pi
```
Changing this to `Math.atan2(stat.eastVelocity, stat.northVelocity)` shows at least one (maybe exactly one? That wayward one we noticed before?) going South.

Reply:
Looks great, will do!